### ISSUE TITLE
Bundle output not created after first compilation

### ISSUE THREAD

**Issue by danigar** [aspect-build/rules_webpack#14]
Hi! Thank you for developing this rule. It looks great ❤️ !

I'm trying to use this rule to speed up builds using webpack and persistent workers.

### Description of the problem
First compilation works great, but any change in source files makes consecutive builds fail because no bundle output is generated.
 
The app I'm trying to build is a typescript and react app (.tsx).

## Minimal reproduction
BUILD file: 

```python
ts_config(
  name = "typescript_config",
  src = "tsconfig.app.json",
  deps = [
    "//:tsconfig.base.json",
    ":tsconfig.json",
  ],
)

SRCS = [
  "src/main.tsx",
  "src/app/app.tsx",
  "src/app/welcome.tsx",
]


DEPS = [
  "@npm//@types",
  "@npm//styled-components",
  "@npm//react",
  "@npm//typescript",
  "@npm//react-dom",
]

ts_project(
  name = "ts",
  tsconfig = ":typescript_config",
  allow_js = True,
  source_map = True,
  srcs = SRCS,
  deps = DEPS,
  supports_workers = True,
)

webpack_bundle(
  name = "bundle",
  data = [
    ":ts",
    ":webpack.config.js",
    "@npm//:node_modules",
  ],
  supports_workers = True,
  entry_point = "src/main.js",
  webpack_config = ":webpack.config.js",
)
```

webpack app config is just a babel loader with some presets:

```
      {
        test: /\.tsx?$/,
        loader: 'babel-loader',
        options: {
          presets: ['@babel/preset-env','@babel/preset-react', '@babel/preset-typescript']
        }
      }
```
Steps:

1. Run `bazel build //:bundle`. First compilation successfully generates the application bundle (using `webpack_dev_server` rule I can see the app up and running)
2. Make a change in one of the ts files (E.g `src/app/welcome.tsx`)
3. Run`bazel build //:bundle` again. It sends the inputs to the webpack worker, worker says that `Compilation Has succeeded` but bazel crashes because no output is generated

## Error
```
output 'apps/examples/example-front/bundle.js' was not created
ERROR: `/`/` Running Webpack [webpack-cli] failed: not all outputs were created or valid
```

## Environment

Output `bazel version`:

```
Build label: 4.2.1
Build target: bazel-out/darwin-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Mon Aug 30 15:24:28 2021 (1630337068)
Build timestamp: 1630337068
Build timestamp as int: 1630337068
```

Tested with bazel 5.0.0 and got the same result.

Rules_nodejs version: `5.1.0`    Rules webpack version: `0.2.1`

**Comment by danigar** [comment]
Debugging the worker a bit seems like everything is ok (I'm not a webpack expert). 

What I've seen is that the `WorkerWatchFileSystem` get the inputs from the `worker_adapter` and it adds the changed source file to `changes map` and `callbackUndelayed`.

**Comment by thesayyn** [comment]
I'll take a look at this at my earliest convenience. 

However, I wanna point out a few things that I feel are wrong but should not affect anything. 

- you might not need preset-typescript as ts_project handles the transpilation part.
- the as above but for preset-react as typescript also handles that part? (not sure though)

things you might try to do whilst we look into this.

1) try to remove babel and leave the transpilation part to typescript
2) disable worker mode for ts_project 
3) disable worker mode for webpack_bundle (just to see if the error is in fact only happens in worker mode.)

**Comment by danigar** [comment]
Thanks to take a look in to this, @thesayyn.

I think you're absolute right about the `preset-typescript`.

Trying out the the options you mentioned:

1. _No babel loader_: It still fails with rebuilds (however first compilation works, so I think there's no need for babel loader).
2. _Disable worker mode for `ts_project`_: Tried this before and it fails too
3. _Disable worker mode for `webpack_bundle`_: Yes, I also tried this one before. The issue does not occur with standalone runs

**Comment by thesayyn** [comment]
thanks for narrowing the possibilities. most certainly something is broken in worker mode.

**Comment by danigar** [comment]
hi @thesayyn. Any updates on this?

**Comment by thesayyn** [comment]
> hi @thesayyn. Any updates on this?

this should be solved by #15. but beware that rules_webpack is not layered on top of rules_nodejs anymore. You need to switch to rules_js in order to use the new version of rules_webpack.

**Comment by thesayyn** [comment]
closing as fixed. feel free to reopen if the issue persists.
